src/App.tsx

```diff
@@ -1,10 +1,69 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { ErrorType } from './types/ErrorType';
+import { FilterBy } from './types/FilterBy';
 
-const USER_ID = 0;
+const USER_ID = 6481;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(ErrorType.none);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [filterBy, setFilterBy] = useState(FilterBy.all);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        setHasError(false);
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setHasError(true);
+        setErrorMessage(ErrorType.load);
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const handleCloseError = () => {
+    setHasError(false);
+  };
+
+  const allTodosCount = todos.length;
+
+  const activeTodosCount = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  useEffect(() => {
+    const filteredTodos = todos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.active:
+          return !todo.completed;
+        case FilterBy.completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
```

Comment:
```
//[HINT] if you need to filter your items by selected filter - use switch/case statement in correct way.
// Perform the necessary filtering only on a case-by-case basis.

// GOOD

const filteredCards = useMemo(() => {
  switch (filter) {
    case CardFilters.All: {
      return cards;
    }

    case CardFilters.Active: {
      return cards.filter(({ deletedDate }) => !deletedDate);
    }

    case CardFilters.Deleted: {
      return cards.filter(({ deletedDate }) => Boolean(deletedDate));
    }

    default:
      return [];
  }
}, [cards, filter]);

// Don't filter your array right away with switch/case inside your callback.
// Because in case CardFilters.All your program just iterate an array and doesn't do any useful actions. So it's a wasting of memory and CPU (very bad)

// BAD

const filteredCards = useMemo(() => {
  return cards.filter(({ deletedDate }) => {
    switch (filter) {
      case CardFilters.All: {
        return true;
      }

      case CardFilters.Active: {
        return !deletedDate;
      }

      case CardFilters.Deleted: {
        return Boolean(deletedDate);
      }

      default:
        return false;
    }
  });
}, [cards, filter]);
```